Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py

```diff
@@ -0,0 +1,290 @@
+        custom_associated_logo = model.custom_logo_associated_domain.get(name, '')
+        if name in model.custom_logo_associated_domain.keys():
+            description = description + ", %s (%s, %s)" % (name, 'Custom Logo', ','.join(custom_associated_logo))
+        else:
+            description = description + ", %s (%s)" % (name, 'Default Logo')
+        logo_list.append(logo)
+    description = description[1:]
+    merged_logos = get_concat_logo_single_image([image_from_base64_to_bytes(logo) for logo in logo_list])
+    res = fileResult(filename=description, data=merged_logos)
+    res['Type'] = entryTypes['image']
+    return_results(res)
+
+
+def main():
+    msg_list = []
+    logo_image_id = demisto.args().get('logoimageId', '')
+    logo_name = demisto.args().get('logoName', '')
+    debug = demisto.args().get('debug', 'False') == 'True'
+    display_logos = demisto.args().get('displayLogos', 'False') == 'True'
+    associated_domains = demisto.args().get('associatedDomains', '').split(',')
+    action = demisto.args().get('action', None)
+
+    if not action and not display_logos:
+        return_error(MSG_NEED_TO_KNOW_WHICH_ACTION)
+        return
+
+    if display_logos:
+        exist, _, _ = oob_model_exists_and_updated()
+        if exist:
+            model = load_demisto_model()
+        else:
+            model = load_model_from_docker()
+        display_all_logos(model)
+        return
+
+    if (action == KEY_ADD_LOGO) and (not logo_image_id or not logo_name):
+        return_error(MSG_EMPTY_NAME_OR_URL)
+    elif not logo_name:
+        return_error(MSG_EMPTY_LOGO_NAME)
+
+    if action==KEY_ADD_LOGO:
+        res = demisto.getFilePath(logo_image_id)
+        path = res['path']
+        with open(path, 'rb') as file:
+            logo_content = file.read()
+
+    exist, demisto_major_version, demisto_minor_version = oob_model_exists_and_updated()
+
+    # Case 1: model in demisto does not exist OR new major released but no logo were added -> load from docker
+    if not exist or (demisto_major_version < MAJOR_VERSION and demisto_minor_version == MINOR_DEFAULT_VERSION):
+        model = load_model_from_docker()
+        if action == KEY_ADD_LOGO:
+            success, msg = model.add_new_logo(logo_name, logo_content, associated_domains)
+        elif action == KEY_REMOVE_LOGO:
+            success, msg = model.remove_logo(logo_name)
+        else:
+            success, msg = model.update_domain_for_custom_logo(logo_name, associated_domains)
+        msg_list.append(msg)
+        if success:
+            minor, model = get_minor_version_upgrade(model)
+            msg = save_upgraded_version_model(model)
+            msg_list.append(msg)
+        else:
+            return_error(msg)
+
+    # Case where there were new new model release -> load model from demisto
+    elif (demisto_major_version == MAJOR_VERSION):
+        model = load_demisto_model()
+        if action == KEY_ADD_LOGO:
```

Comment:
code duplication - seems like line 227-239 and 244-256 are totally the same
you can use the if just to load the model and then preform the extact same logic for adding \ removing logo or just extract this logic to a function

Reply:
not exactly the same but I created 1 function